modules/azure/availabilityset.go

```diff
@@ -0,0 +1,132 @@
+package azure
+
+import (
+	"context"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-07-01/compute"
+	"github.com/gruntwork-io/terratest/modules/collections"
+	"github.com/gruntwork-io/terratest/modules/testing"
+	"github.com/stretchr/testify/require"
+)
+
+// AvailabilitySetExists indicates whether the speficied Azure Availability Set exists
+func AvailabilitySetExists(t testing.TestingT, avsName string, resGroupName string, subscriptionID string) bool {
+	exists, err := AvailabilitySetExistsE(t, avsName, resGroupName, subscriptionID)
+	require.NoError(t, err)
+	return exists
+}
+
+// AvailabilitySetExistsE indicates whether the speficied Azure Availability Set exists
+func AvailabilitySetExistsE(t testing.TestingT, avsName string, resGroupName string, subscriptionID string) (bool, error) {
+	_, err := GetAvailabilitySetE(t, avsName, resGroupName, subscriptionID)
+	if err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+// CheckAvailabilitySetContainsVM checks if the Virtual Machine is contained in the Availability Set VMs
+func CheckAvailabilitySetContainsVM(t testing.TestingT, vmName string, avsName string, resGroupName string, subscriptionID string) bool {
+	avsVMs, err := GetAvailabilitySetVMsE(t, avsName, resGroupName, subscriptionID)
+	if err != nil {
+		return false
+	}
+
+	return collections.ListContains(avsVMs, strings.ToLower(vmName))
```

Comment:
Why `strings.ToLower`? Is it all case insensitive?

Reply:
The resource ID in these calls come all CAPS.. I changed the method name to reflect that so there is no ambiguity and removed the lowercase conversion.